CoreGraphUtilities/src/au/gov/asd/tac/constellation/graph/utilities/hashmod/Hashmod.java

```diff
@@ -0,0 +1,171 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.graph.utilities.hashmod;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import static javatests.TestSupport.fail;
+
+/**
+ * A text hashmod based on a supplied CSV file.  Will modify attributes
+ * specified in the headers to be values based on the first Key column.
+ *
+ * @author CrucisGamma
+ */
+public class Hashmod {
+
+    public static final String ATTRIBUTE_NAME = "hashmod";
+    private HashmodCSVImportFileParser parser;
+    private String csvFileStr;
+    private List<String[]> data;
+
+    public Hashmod() {
+        parser = null;
+        csvFileStr = "";
+    }
+
+    /**
+     * Create a new Hashmod.
+     *
+     * @param csvFile Name of the CSV file the user has chosen
+     */
+    public Hashmod(final String csvFileStr) {
+        if (csvFileStr == null) {
+            this.csvFileStr = "";
+        } else {
+            this.csvFileStr = csvFileStr;
+        }
+
+        parser = new HashmodCSVImportFileParser();
+        try {
+            data = parser.parse(new HashmodInputSource(new File(csvFileStr)), null);
+        } catch (IOException ex) {
+            fail("IO Exception : " + ex.getLocalizedMessage());
+        }
+    }
+
+    public void setCSVFileStr(final String csvFileStr) {
+        this.csvFileStr = csvFileStr;
+        parser = new HashmodCSVImportFileParser();
+        try {
+            data = parser.parse(new HashmodInputSource(new File(csvFileStr)), null);
+        } catch (IOException ex) {
+            fail("IO Exception : " + ex.getLocalizedMessage());
+        }
+    }
+
+    public String[] getCSVFileHeaders() {
+        if (data != null && data.size() > 0) {
+            return data.get(0);
+        }
+        return null;
+    }
+
+    public String[] getCSVRow(final int row) {
+        if (data != null && data.size() > row) {
+            return data.get(row);
+        }
+        return null;
+    }
+
+    public String getCSVKey() {
+        String[] headers = getCSVFileHeaders();
+        if (headers != null && headers.length > 0) {
+            if (headers.length > 0) {
+                return headers[0];
+            }
+        }
+        return null;
+    }
+
+    public HashMap<String, Integer> getCSVKeys() {
+        final HashMap<String, Integer> keys = new HashMap<>();
+        if (data != null && data.size() > 0) {
+            int i;
+            for (i = 1; i < data.size(); i++) {
```

Comment:
In Java, you don't actually need to declare the loop variable beforehand. Change the above 2 lines to:
`for (int i = 1; i < data.size(); i++) {`